lib/rspec/matchers/built_in/all.rb

```diff
@@ -50,7 +50,13 @@ def match(_expected, _actual)
         def index_failed_objects
           actual.each_with_index do |actual_item, index|
             cloned_matcher = matcher.clone
-            matches = cloned_matcher.matches?(actual_item)
+
+            begin
+              matches = cloned_matcher.matches?(actual_item)
+            rescue StandardError
+              matches = nil
```

Comment:
Ok. Given an exception is raised by using a matcher, say

```ruby
it { expect([1, 'a', 11]).to any(be < 10) }
```

This will stop evaluation of matchers immediately. `11` is not checked anymore.